tests/common/environment.rs

```diff
@@ -54,6 +61,111 @@ impl TestEnvironment {
     pub async fn call(&self, req: actix_http::Request) -> ServiceResponse {
         self.test_app.call(req).await.unwrap()
     }
+
+    pub async fn generate_friend_user_notification(&self) {
+        let resp = self
+            .add_user_to_team(
+                &self.dummy.as_ref().unwrap().alpha_team_id,
+                FRIEND_USER_ID,
+                USER_USER_PAT,
+            )
+            .await;
+        assert_status(resp, StatusCode::NO_CONTENT);
+    }
+
+    pub async fn remove_project(&self, project_slug_or_id: &str, pat: &str) -> ServiceResponse {
```

Comment:
I think the helper functions here on `TestEnvironment`  is a really good idea for avoiding duplicated code in setting up tests.

This is probably going to change a lot though soon once we have v3 routes though. Maybe we can do two `TestEnvironments` (or a struct in the fields of `TestEnvironment` that contains the helpers) that implement a common trait?

Reply:
Thanks!

And after some separate discussion I'll be changing it to put the helpers into a separate struct that's accessible through the TestEnvironment.